submissions/Nik3264/oop-exercise/index.js

```diff
@@ -0,0 +1,163 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _https://github.com/Nik3264/OOP-exercise_
+   Web app: _https://nik3264.github.io/OOP-exercise/_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Animal {
+  constructor(species, name, gender, legs, saying) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.saying = saying;
+    this.properties = ["species", "name", "gender", "legs", "saying"];
```

Comment:
The child classes inherit this property but never populate it, so it keeps containing the list of the root class' properties. This makes it useless outside root class and not worth inheriting.
You could use this array of property names as a starting array in `getProperties` method to map them into the method's output.